pkg/operator/resourcesynccontroller/core.go

```diff
@@ -55,18 +61,17 @@ func CombineCABundleConfigMaps(destinationConfigMap ResourceLocation, lister cor
 
 	caBytes, err := crypto.EncodeCertificates(finalCertificates...)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
-	cm := &corev1.ConfigMap{
-		ObjectMeta: certrotation.NewTLSArtifactObjectMeta(
-			destinationConfigMap.Name,
-			destinationConfigMap.Namespace,
-			additionalAnnotations,
-		),
-		Data: map[string]string{
-			"ca-bundle.crt": string(caBytes),
-		},
+	modified := additionalAnnotations.EnsureTLSMetadataUpdate(&cm.ObjectMeta)
+	newCMData := map[string]string{
+		"ca-bundle.crt": string(caBytes),
+	}
+	if !reflect.DeepEqual(cm.Data, newCMData) {
+		modified = true
 	}
-	return cm, nil
+
+	cm.Data = newCMData
```

Comment:
Nit: I expected to see the assignment to cm.Data inside the "then" block with the assignment to modified. It looks correct to me either way.